File: backend/src/main/java/com/backend/ord/services/AuthenticationService.java
package com.backend.ord.services;

import com.backend.ord.api.requests.LoginRequest;
import com.backend.ord.api.requests.RegisterRequest;
import com.backend.ord.domain.entities.User;
import com.backend.ord.exceptions.ForbiddenException;
import com.backend.ord.exceptions.UserNotFoundException;
import jakarta.servlet.http.HttpServletRequest;
import jakarta.servlet.http.HttpServletResponse;

public interface AuthenticationService {
    User register(
            RegisterRequest request,
            HttpServletResponse response
    ) throws UserNotFoundException;

    User login(
            LoginRequest request,
            HttpServletResponse response
    ) throws UserNotFoundException;

    void logout(HttpServletRequest request, HttpServletResponse response) throws ForbiddenException;
}

File: backend/src/main/kotlin/com/backend/ord/services/AuthenticationService.kt
package com.backend.ord.services

import com.backend.ord.api.requests.LoginRequest
import com.backend.ord.api.requests.RegisterRequest
import com.backend.ord.core.user.model.UserEntity
import com.backend.ord.exceptions.ForbiddenException
import com.backend.ord.exceptions.UserNotFoundException
import jakarta.servlet.http.HttpServletRequest
import jakarta.servlet.http.HttpServletResponse

interface AuthenticationService {
    @Throws(UserNotFoundException::class)
    fun register(
        request: RegisterRequest,
        response: HttpServletResponse
    ): UserEntity

    @Throws(UserNotFoundException::class)
    fun login(
        request: LoginRequest,
        response: HttpServletResponse
    ): UserEntity

    @Throws(ForbiddenException::class)
    fun logout(request: HttpServletRequest, response: HttpServletResponse)
}
